.github/workflows/build-and-test.yml

```
@@ -139,3 +139,167 @@ jobs:
+    name: cross-compile
+    runs-on: ubuntu-latest
+    needs: [setup-environment]
+    strategy:
+      matrix:
+        SYS_BINARIES: [ "binaries-darwin_amd64", "binaries-linux_amd64", "binaries-linux_arm64", "binaries-windows_amd64" ]
+    steps:
+      - name: Check out the codebase.
+        uses: actions/checkout@v2
+
+      - name: Set up Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ env.GO_VERSION }}
+
+      - name: Caching dependency
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cache/go-build
+            ~/go/pkg/mod
+            ~/go/bin
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+
+      - name: Build Collector
+        run: |
+          make ${{ matrix.SYS_BINARIES }}
+      
+      - name: Updating binaries
+        uses: actions/upload-artifact@v2
+        with:
+          name: ${{ matrix.SYS_BINARIES }}
+          path: |
+            ./bin/*
+
+  windows-msi:
+    name: windows-msi
+    runs-on: ubuntu-latest
+    needs: [cross-compile]
+    steps:
+      - name: Check out the codebase.
+        uses: actions/checkout@v2
+
+      - name: Downloading binaries-windows_amd64
+        uses: actions/download-artifact@v2
+        with:
+          name: binaries-windows_amd64
+          path: ./bin
+
+      - name: Caching dependency
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cache/go-build
+            ~/go/pkg/mod
+            ~/go/bin
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+
+      - name: Build MSI
+        run: |
+          mkdir -p dist
+          gitRef=(${GITHUB_REF//// })
+          if [[ "${gitRef[1]}" = "tags" ]] ;
+          then
+            echo "TAG_NAME=${gitRef[2]}" >> $GITHUB_ENV
+          else
+            echo "TAG_NAME=" >> $GITHUB_ENV
+          fi
+          export VERSION_TAG="${TAG_NAME#v}"
+          make msi SKIP_COMPILE=true VERSION="${VERSION_TAG:-0.0.1.${{ github.run_number }}}"
+      
+      - name: Uploading msi build artifacts
+        uses: actions/upload-artifact@v2
+        with:
+          name: msi-build
+          path: ./dist/*.msi
+
+  windows-msi-validation:
+    name: windows-msi-validation
+    runs-on: windows-latest
+    needs: [windows-msi]
+    strategy:
+      matrix:
+        MODE: [ "agent", "gateway" ]
+    steps:
+      - name: Check out the codebase.
+        uses: actions/checkout@v2
+
+      - name: Downloading msi build
+        uses: actions/download-artifact@v2
+        with:
+          name: msi-build
+          path: ./dist
+
+      - name: Installation test
+        run: |
+          $ErrorActionPreference = 'Stop'
+          Set-PSDebug -Trace 1
+          $msi_path = Resolve-Path .\dist\splunk-otel-collector*.msi
+          $env:VERIFY_ACCESS_TOKEN = "false"
+          .\internal\buildscripts\packaging\installer\install.ps1 -access_token "testing123" -msi_path "$msi_path" -mode "${{ matrix.MODE }}"
```

```suggestion
          .\internal\buildscripts\packaging\installer\install.ps1 -access_token "testing123" -realm "test" -msi_path "$msi_path" -mode "${{ matrix.MODE }}"
```
Please use a fake realm to not spam our production realm.

---

Yes based on your above suggestion I have checked that default value for realm us0 which would be by default picked up if we don’t specify realm and adding a fake realm would avoid those spamming of the production with the dummy requests. Hence adding a fake realm is a nice idea and I have added that (It was not existing in the CircleCI job but I have added it in the GitHub Action workflow now). Thanks for the suggestion!